### PR TITLE
feat(pubsub): implement schema admin examples 1/2

### PR DESCRIPTION
**Pull request by coryan**
Part of the work for #5706

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5914)

### CODE REVIEW

File: google/cloud/pubsub/samples/samples.cc [hunk 1]
```diff
@@ -669,6 +675,171 @@ void ExampleStatusOr(google::cloud::pubsub::TopicAdminClient client,
...
+     std::string const& schema_id) {
+    auto constexpr kDefinition = R"pfile(
+syntax = "proto3";
+package google.cloud.pubsub.samples;
+
+message State {
+  string name = 1;
+  string post_abbr = 2;
+}
+)pfile";
```
Comment (devbww) [inline google/cloud/pubsub/samples/samples.cc:729]: Can we indent?

File: google/cloud/pubsub/samples/samples.cc [hunk 2]
```diff
@@ -669,6 +675,171 @@ void ExampleStatusOr(google::cloud::pubsub::TopicAdminClient client,
...
+        {
+          "name": "post_abbr",
+          "type": "string",
+          "doc": "The postal code abbreviation of the state."
+        }
+      ]
+    })js";
+    auto schema = client.CreateAvroSchema(
+        experimental::Schema(project_id, schema_id), kDefinition);
+    if (!schema) throw std::runtime_error(schema.status().message());
```
Comment (devbww) [inline google/cloud/pubsub/samples/samples.cc:705]: Were these "create" calls cases where we should document tolerating "already exists"?

**Review by devbww — APPROVED** [review]

File: google/cloud/pubsub/samples/samples.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/pubsub/samples/samples.cc:705]: yes, fixed. Also fixed the `DeleteSchema()` and `kNotFound` case.

File: google/cloud/pubsub/samples/samples.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/pubsub/samples/samples.cc]: To my surprise, yes.

**Review by devbww — APPROVED** [review]

**Pull request merged**